### PR TITLE
refactor: page view context communication support

### PR DESCRIPTION
**Pull request by yandiaszup**
## Description

*Since we have launched Communication between Components (#54), we can now update the way PageView and PageIndicator communicates with each other by starting to use Context, Expressions, and Actions.*

## Related Issues

*#422*

## Tests

I added the following tests:

IOS
Added tests to PageIndicator: `PageIndicatorTests`
Remove PageView tests with page indicator as a atribute of PageView : `test_viewWithPagesAndIndicator()`
Remove PageViewUIComponent test that was testing the direct communication between PageView and PageIndicator: 
`test_whenCallingSwipePage_thenShouldUpdateModel`

Android
Remove `buildView should call addView when indicator is not null` on PageViewTest
Change `toView_should_return_BeaglePageIndicatorView_and_set_colors` to  `toView_should_return_BeaglePageIndicatorView_set_colors_and_setCount` on PageIndicatorTest
Remove `setCount_should_call_BeaglePageIndicatorView_setCount` on PageIndicatorTest
Remove `make_should_return_moshi_to_serialize_a_UndefinedComponent_of_type_PageIndicatorComponent` on BeagleMoshiTest
Add `when_bind_change_should_call_onItemUpdated` on PageIndicatorTest

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [DCO].
- [x] All existing and new tests are passing.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*

[issue database]: https://github.com/ZupIT/beagle/issues
[DCO]: https://developercertificate.org/

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/main/java/br/com/zup/beagle/android/data/serializer/adapter/ComponentJsonAdapterFactory.kt [hunk 1]
```diff
@@ -65,8 +64,10 @@ internal object ComponentJsonAdapterFactory {
     private fun registerBaseSubTypes(
         factory: PolymorphicJsonAdapterFactory<ServerDrivenComponent>
     ): PolymorphicJsonAdapterFactory<ServerDrivenComponent> {
-        return factory.withBaseSubType(PageIndicatorComponent::class.java)
-            .withBaseSubType(InputWidget::class.java)
+//        return factory.withBaseSubType(PageIndicatorComponent::class.java)
```
Comment (danilorochazup) [inline android/beagle/src/main/java/br/com/zup/beagle/android/data/serializer/adapter/ComponentJsonAdapterFactory.kt:67]: These comments may be unnecessary, what do you think?

Comment (viniciusguardieirozup) [reply android/beagle/src/main/java/br/com/zup/beagle/android/data/serializer/adapter/ComponentJsonAdapterFactory.kt:67]: I forgot to remove

File: iOS/Sources/Beagle/Sources/Components/Page+Renderable/PageViewUIComponent.swift [hunk 2]
```diff
@@ -69,6 +69,10 @@ class PageViewUIComponent: UIView {
     }
 
     // MARK: - Subviews
+    
+    override func sizeThatFits(_ size: CGSize) -> CGSize {
+        .init(width: size.width, height: 40)
+    }
```
Comment (lucasaraujo) [inline iOS/Sources/Beagle/Sources/Components/Page+Renderable/PageViewUIComponent.swift:75]: Why the height 40 here?

Comment (yandiaszup) [reply iOS/Sources/Beagle/Sources/Components/Page+Renderable/PageViewUIComponent.swift]: There is no need for this function in this class, i just remove it.

**Comment by theffc** [comment]
We still need to have the same functionality of PageView and PageIndicator that we had before these changes, that is, we still need that PageView and PageIndicator work without using context. So, we need to think in a way of doing that while explicitly saying that's something **deprecated**.

**Comment by yandiaszup** [comment]
Closing this Pull Request because we will use a different approach to this PageView refactor

**Pull request closed without merging**